adLaCarteHistory.go

```diff
@@ -1,114 +1,83 @@
+	"fmt"
+	"io"
 	"io/ioutil"
 	"log"
 	"net/http"
+	"os"
 	"strconv"
 	"strings"
-	"time"
-)
-
-var CREDENTIALS string
 
-const URL_API = "https://adlacarte.adesso.de/api/"
-const URL_LOGIN = URL_API + "login"
-const URL_SUPPLIERS = URL_API + "offices/1/suppliers/"
-const URL_REACHED_ORDER_VALUE = "/reachedOrderValue"
+	"gopkg.in/resty.v1"
+)
 
-var suppliersToName = [7]string{"Entenhaus", "ChiliPeppers", "PiDoe", "ChinaImbissBUI", "PizzeriaMammaMia", "Pinnochio", "PizzariabeiMarco"}
+type Supplier struct {
+	id                int
+	name              string
+	office            int
+	reachedOrderValue float32
+}
 
-var suppliersToGauge [7]prometheus.Gauge
+var apiURL = getEnv("ADLACARTE_API_URL", "https://adlacarte.adesso.de/api/")
 
 func main() {
-	CREDENTIALS = readCredentialsFromFile("credentials")
-	http.Handle("/metrics", promhttp.Handler())
-
-	for i := 0; i < len(suppliersToGauge); i++ {
-		suppliersToGauge[i] = prometheus.NewGauge(
-			prometheus.GaugeOpts{
-				Namespace: "adLaCarte",
-				Name:      suppliersToName[i],
-			})
-		prometheus.MustRegister(suppliersToGauge[i])
+	var suppliers = []Supplier{
+		Supplier{0, "Entenhaus", 1, 0.0},
+		Supplier{1, "Chili Peppers", 1, 0.0},
+		Supplier{2, "PIDÖ", 1, 0.0},
+		Supplier{3, "China Imbiss BUI", 1, 0.0},
+		Supplier{4, "Pizzeria Mamma Mia", 1, 0.0},
+		Supplier{5, "Pinnochio", 1, 0.0},
+		Supplier{6, "Pizzeria bei Marco", 1, 0.0},
 	}
-	client := resty.New()
-
-	go func() {
-		// Console output
-		line := ""
-		for i := 0; i < len(suppliersToName); i++ {
-			line += "| " + suppliersToName[i] + " "
+	client := buildClient()
+	http.HandleFunc("/metrics", func(w http.ResponseWriter, r *http.Request) {
+		for _, supplier := range suppliers {
+			if e := fetchMetricsForSupplier(&supplier, client); e != nil {
+				log.Println(e)
+			}
 		}
-		log.Println(line + "|")
-
-		for true {
-			updateValues(client)
-			time.Sleep(15 * time.Second)
-		}
-	}()
+		writeMetrics(suppliers[:], w)
+	})
 	log.Fatal(http.ListenAndServe(":8080", nil))
 }
 
-func updateValues(client *resty.Client) {
-	line := ""
-	for i := 0; i < len(suppliersToGauge); i++ {
-		value := getValueOf(i+1, client)
-		suppliersToGauge[i].Set(value)
-
-		// Console output
-		line += "| "
-		valueAsString := strconv.FormatFloat(value, 'f', 2, 64)
-		if valueAsString == "0" {
-			valueAsString = "0.00"
-		}
-		length := len(suppliersToName[i]) - len(valueAsString)
-		for j := 0; j < length; j++ {
-			line += " "
-		}
-		line += valueAsString + " "
+func fetchMetricsForSupplier(supplier *Supplier, client *resty.Client) error {
+	response, e := client.R().Get(fmt.Sprintf("%s/offices/%d/suppliers/%d", apiURL, supplier.office, supplier.id))
+	if e != nil {
+		return e
 	}
-	log.Println(line + "|")
-}
-
-func authorize(client *resty.Client) {
-	_, e := client.R().
-		SetHeader("authorization", "Basic "+CREDENTIALS).
-		Get(URL_LOGIN)
+	responseAsString := string(response.Body())
+	floatVal, e := strconv.ParseFloat(responseAsString, 32)
```

Comment:
Right now you are parsing a json object containing information about the supplier. You probably wanted to query `/reachedOrderValue`. The endpoint returns an integer, you have to add the decimal point manually, see https://github.com/JulianSauer/adLaCarteHistory/blob/master/adLaCarteHistory.go#L102